samples/40.timex-resolution/src/main/java/com/microsoft/bot/sample/timex/resolution/Ambiguity.java

```diff
@@ -0,0 +1,147 @@
+
+        /**
+         * We should find two results in this example.
+         */
+        for (ModelResult result : results)
+        {
+            /**
+             * The resolution includes two example values: going backwards and forwards from NOW in the calendar.
+             */
+            HashSet<String> distinctTimexExpressions = new HashSet<String>();
+            List<Dictionary<String, String>> values = (List<Dictionary<String, String>>)result.resolution.get("values");
+            for (Dictionary<String, String> value : values)
+            {
+                /**
+                 * Each result includes a TIMEX expression that captures the inherent date but not time ambiguity.
+                 * We are interested in the distinct set of TIMEX expressions.
+                 */
+                if (value.get("timex") != null)
+                {
+                    String timex = value.get("timex");
+                    distinctTimexExpressions.add(timex);
+                }
+                /**
+                 * There is also either a "value" property on each value or "start" and "end".
+                 * If you use ToString() on a TimeProperty object you will get same "value".
+                 */
+            }
+
+            /**
+             * The TIMEX expression captures date ambiguity so there will be a single distinct expression for each result.
+             */
+            String string = String.join(result.text,",", distinctTimexExpressions.toString());
+            System.out.println(string);
+
+            /**
+             * The result also includes a reference to the original string - but note the start and end index are both inclusive.
+             */
+            }
+    }
+
+    public static void timeAmbiguity()
+    {
+        /**
+         * Run the recognizer.
+         */
+        List<ModelResult> results = DateTimeRecognizer.recognizeDateTime("We would like to arrive at 4 o'clock or 5 o'clock.", Culture.English);
+
+        /**
+         * We should find two results in this example.
+         */
+        for (ModelResult result : results)
+        {
+            /**
+             * The resolution includes two example values: one for AM and one for PM.
+             */
+            HashSet<String> distinctTimexExpressions = new HashSet<String>();
+            List<Dictionary<String, String>> values = (List<Dictionary<String, String>>)result.resolution.get("values");
+            for (Dictionary<String, String> value : values)
+            {
+                /**
+                 * Each result includes a TIMEX expression that captures the inherent date but not time ambiguity.
+                 * We are interested in the distinct set of TIMEX expressions.
+                 */
+                if (value.get("timex") != null)
+                {
+                    String timex = value.get("timex");
```

Comment:
Same of the `timex` mentioned above

Reply:
The problem is that we need to validate if the value "timex" exists before trying to get it, If we do this outside the if I think it will break everytime that the value doesn't exist